sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -282,316 +479,363 @@ q.parallel_for(nd_range<2>(G, L), [=](nd_item<2> item)
-  using type_b = Tb; // this type alias is not available in the current implementation
-  using type_accumulator = Tc; // this type alias is not available in the current implementation
-
-  // construct the matrices using the default sizes
-  static constexpr std::size_t M = 16;
-  static constexpr std::size_t N = 16;
-  static constexpr std::size_t K = ((sizeof(Ta) == 1) ? 64 : 32);
-
-  template <typename Group, layout LayoutA>
-  using joint_matrix_a = joint_matrix<Group, Ta, use::a, M, K, LayoutA>;
-  template <typename Group, layout LayoutB>
-  using joint_matrix_b = joint_matrix<Group, Tb, use::b, K, N, LayoutB>;
   template <typename Group>
-  using joint_matrix_accumulator = joint_matrix<Group, Tc, use::accumulator, M, N>;
-
-  static constexpr uint32_t numtiles = 8;
-  static constexpr scope_t scopes[] = {scope_t::sub_group};
-  static constexpr int num_scopes = sizeof(scopes) / sizeof(scope_t);
-  struct combination {
-    uint32_t max_msize;
-    uint32_t max_nsize;
-    uint32_t max_ksize;
-    uint32_t msize;
-    uint32_t nsize;
-    uint32_t ksize;
-    matrix_type atype;
-    matrix_type btype;
-    matrix_type accumulatortype;
-  };
-  // In this case, the combinations array contain only the combinations that correspond to the Ta, Tb, and Tc 
-  // types that the user provided
-  static constexpr combination combinations[] = {
-      {16, 16, (sizeof(Ta) == 1) ? 64 : 32}};
-  static constexpr int num_combinations =
-      sizeof(combinations) / sizeof(combination);
+  using joint_matrix_d = joint_matrix<Group, Td, use::accumulator, sM, sN>;
 };
 
-// General query form:
-// types are not given, no default sizes and no implicit matrix construction
-template <int sM, int sN, int sK>
-struct tpu_params<tpu::amx, void, void, void, sM, sN, sK> {
-  static constexpr uint32_t numtiles = 8;
-  static constexpr scope_t scopes[] = {scope_t::sub_group};
-  static constexpr int num_scopes = sizeof(scopes) / sizeof(scope_t);
-  struct combination {
-    uint32_t max_msize;
-    uint32_t max_nsize;
-    uint32_t max_ksize;
-    uint32_t msize;
-    uint32_t nsize;
-    uint32_t ksize;
-    matrix_type atype;
-    matrix_type btype;
-    matrix_type accumulatortype;
-  };
-  
-  static constexpr combination combinations[] = {
-      {16, 16, 64, 0, 0, 0, matrix_type::sint8, matrix_type::sint8, matrix_type::sint32},
-      {16, 16, 64, 0, 0, 0, matrix_type::sint8, matrix_type::uint8, matrix_type::sint32},
-      {16, 16, 64, 0, 0, 0, matrix_type::uint8, matrix_type::sint8, matrix_type::sint32},
-      {16, 16, 64, 0, 0, 0, matrix_type::uint8, matrix_type::uint8, matrix_type::sint32},
-      {16, 16, 32, 0, 0,0, matrix_type::bf16, matrix_type::bf16, matrix_type::fp32}};
-  static constexpr int num_combinations =
-      sizeof(combinations) / sizeof(combination);
-};
+// This is the default values form, where the matrix dimensions are
+// omitted.
+template<architecture Arch, typename Ta, typename Tb, typename Tc, typename Td=Tc>
```

Comment:
```suggestion
template<architecture Arch, typename Ta, typename Tb, typename Tc, typename Td>
```

C++ does not allow you to provide a default template parameter for a template specialization, so the `=Tc` part will cause a compilation failure.  If you want `Td` to have a default value, it must be specified in the primary template (i.e. the first one defined in the synopsis).

Reply:
I think you are looking at the wrong synopsis. 
The current version is missing the primary template so I am adding that. I will specify the default there.